common/channel/src/libra_channel.rs

```diff
@@ -0,0 +1,151 @@
+use std::collections::{HashMap, VecDeque};
+use std::pin::Pin;
+use std::sync::{Arc, Mutex};
+use std::task::Waker;
+
+use futures::async_await::FusedStream;
+use futures::stream::Stream;
+use futures::task::Context;
+use futures::Poll;
+
+/// ValidatorMessage trait needs to be implemented by every message which gets pushed into the Libra
+/// channel so that we can ensure fairness among validators
+pub trait ValidatorMessage {
+    /// Extract the Validator from which this message arrived. This
+    /// will be used for ensuring fairness among validators.
+    fn get_validator(&self) -> &str;
+}
+
+/// Type is an enum which can be used as a configuration option for
```

Comment:
nit: Could you extend the comments and explain the policy for treating the messages when the queue is full depending on the policy? (drop the oldest message in the queue if it's LIFO / drop the new message if it's FIFO).

Reply:
Done